addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java

```diff
@@ -203,44 +204,85 @@ public void initialize() {
-                logger.debug("Cann't connect to Freebox server {}", e);
+            boolean httpsRequestOk = false;
+            if (configuration.useOnlyHttp == null || configuration.useOnlyHttp.equals(Boolean.FALSE)) {
+                try {
+                    result = HttpUtil.executeUrl("GET", "https://" + configuration.fqdn + "/api_version", 5000);
+                    httpsRequestOk = true;
+                } catch (IOException e) {
+                    logger.debug("Cann't connect to {} with HTTPS: {}", configuration.fqdn, e);
+                }
             }
-            if (result != null) {
+            if (result == null) {
+                try {
+                    result = HttpUtil.executeUrl("GET", "http://" + configuration.fqdn + "/api_version", 5000);
+                } catch (IOException e) {
+                    logger.debug("Cann't connect to {} with HTTP: {}", configuration.fqdn, e);
+                }
+            }
+            String apiBaseUrl = null;
+            String apiVersion = null;
+            String hardwareVersion = null;
+            String apiDomain = null;
+            boolean httpsAvailable = false;
+            String httpsPort = null;
+            boolean useHttps = false;
+            String fqdn = configuration.fqdn;
+            String errorMsg = null;
+            if (result == null) {
+                errorMsg = "Cann't connect to " + configuration.fqdn;
+            } else {
                 apiBaseUrl = StringUtils.trim(StringUtils
                         .replace(StringUtils.substringBetween(result, "\"api_base_url\":\"", "\""), "\\/", "/"));
                 apiVersion = StringUtils.trim(StringUtils.substringBetween(result, "\"api_version\":\"", "\""));
                 hardwareVersion = StringUtils.trim(StringUtils.substringBetween(result, "\"device_type\":\"", "\""));
-                // apiDomain = StringUtils.trim(StringUtils.substringBetween(result, "\"api_domain\":\"", "\""));
-                // httpsAvailable = StringUtils.containsIgnoreCase(result, "\"https_available\":true");
-                // httpsPort = StringUtils.trim(StringUtils.substringBetween(result, "\"https_port\":", ","));
+                apiDomain = StringUtils.trim(StringUtils.substringBetween(result, "\"api_domain\":\"", "\""));
+                httpsAvailable = StringUtils.containsIgnoreCase(result, "\"https_available\":true");
+                httpsPort = StringUtils.trim(StringUtils.substringBetween(result, "\"https_port\":", ","));
+                if (apiBaseUrl == null || apiBaseUrl.isEmpty()) {
+                    errorMsg = configuration.fqdn + " does not deliver any API base URL";
+                } else if (apiVersion == null || apiVersion.isEmpty()) {
+                    errorMsg = configuration.fqdn + " does not deliver any API version";
+                } else if (httpsAvailable
+                        && (configuration.useOnlyHttp == null || configuration.useOnlyHttp.equals(Boolean.FALSE))) {
```

Comment:
If you change it to  Boolean.FALSE.equals(configuration.useOnlyHttp) you don't need the null check.